tests/unit/multipass/test_multipass_instance.py

```diff
@@ -573,6 +607,91 @@ def test_push_file_no_parent_directory(mock_multipass, instance, tmp_path):
     assert str(exc_info.value) == "Directory not found: '/tmp'"
 
 
+def test_push_file_mktemp_error(mock_multipass, instance, tmp_path):
+    """Raise an error if the temporary file cannot be created."""
+    error = subprocess.CalledProcessError(-1, ["mktemp"], "test stdout", "test stderr")
+
+    mock_multipass.exec.side_effect = [
+        # test call
+        mock.Mock(returncode=0),
+        # mktemp call
+        error,
+    ]
+
+    source = tmp_path / "src.txt"
```

Comment:
you can probably create a small `test_file` fixture local to this test module to avoid this repetition

Reply:
Done!